lz4/blob_compressor.go

```diff
@@ -0,0 +1,62 @@
+package lz4
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/pierrec/lz4/v4"
+)
+
+type BlobCompressor struct {
+	prefix        []byte
+	prefixPlusLen int
+	limit         int
+}
+
+var defaultPrefix = []byte{0x01, 0x11, 0x22, 0x33}
+
+func NewBlobCompressor(limit int) BlobCompressor {
+	return BlobCompressor{
+		prefix:        defaultPrefix,
+		prefixPlusLen: len(defaultPrefix) + 4,
+		limit:         limit,
+	}
+}
+
+func (c BlobCompressor) Compress(data []byte) ([]byte, error) {
+	if len(data) < c.limit {
+		return data, nil
+	}
+	buf := make([]byte, len(c.prefix)+lz4.CompressBlockBound(len(data)+4))
+	copy(buf, c.prefix)
+
```

Comment:
This +4 looks like a mistake, we compress the data with size `len(data)`, not `len(data)+4`